### DISCUSSION TITLE
Brainstorm: A NuOS

### DISCUSSION THREAD

**Discussion by dch82** [nushell/nushell#9757] [Ideas]
What if we extended the Nu philosophy onto a new OS free of POSIX and other legacy standards?

Comment your ideas down below!

**Comment by hardfau1t** [comment]
What do you mean by os? Is it kernel or is it userspace environment? If it is kernel i think its a bad idea because for users most thing that matters is the day to day application

**Reply by dch82** [reply]
Also we don't need another [GNU Hurd](https://en.m.wikipedia.org/wiki/GNU_Hurd)

**Comment by dch82** [comment]
Mostly userspace (we need a Nushell-friendly coreutils library), but we can implement ideas like an extra pipe for structured output so it doesn't pass through stdout/in.(Also I want to use cool and useful ideas which popped up after Unix)

**Reply by dch82** [reply]
Should we code new utils from scratch or just create wrappers around them with [`jc`](https://github.com/kellyjonbrazil/jc) and `from json`?
